### DISCUSSION TITLE
Issue with nhl_api_client not parsing dates correctly?

### DISCUSSION THREAD

**Discussion by rafuzo** [riffnshred/nhl-led-scoreboard#445] [General]
I got to the point where I have everything set up and the matrix renders nicely; however I noticed my team's score was not coming through while I watched the game. I noticed this in my debug logs:

`24-01-15 16:08:08 INFO HTTP Request: GET https://api-web.nhle.com/v1/score/2024-1-15 "HTTP/1.1 404 Not Found"`

I tried to curl this from the cmdline and also got a 404; however, when I curl with padded two-digits in the month and year, I get data back:

> ~ % wget https://api-web.nhle.com/v1/score/2024-1-15
> --2024-01-15 16:09:28--  https://api-web.nhle.com/v1/score/2024-1-15
> Resolving api-web.nhle.com (api-web.nhle.com)... 2606:4700::6813:972a, 2606:4700::6813:982a, 104.19.152.42, ...
> Connecting to api-web.nhle.com (api-web.nhle.com)|2606:4700::6813:972a|:443... connected.
> HTTP request sent, awaiting response... 404 Not Found
> 2024-01-15 16:09:28 ERROR 404: Not Found.
> 
> ~ % wget https://api-web.nhle.com/v1/score/2024-01-15
> --2024-01-15 16:09:34--  https://api-web.nhle.com/v1/score/2024-01-15
> Resolving api-web.nhle.com (api-web.nhle.com)... 2606:4700::6813:982a, 2606:4700::6813:972a, 104.19.152.42, ...
> Connecting to api-web.nhle.com (api-web.nhle.com)|2606:4700::6813:982a|:443... connected.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [application/json]
> Saving to: ‘2024-01-15’
> 
> 2024-01-15                                                                 [ <=>                                                                                                                                                                         ]  27.21K  --.-KB/s    in 0.001s
> 
> 2024-01-15 16:09:34 (31.9 MB/s) - ‘2024-01-15’ saved [27865]

Is anyone else encountering an issue like this?

**Comment by pjockey** [comment]
Perhaps read this https://github.com/riffnshred/nhl-led-scoreboard/pull/440 This might be why.
